File: server/src/main/java/server/ParkingLot.java
package server;

import java.util.Arrays;

import com.google.gson.Gson;

public class ParkingLot {

	private final int width;
	private final int height;
	private String[] lot;
	
	private Gson gson = new Gson();
	
	public ParkingLot(int w, int h) {
		width = w;
		height = h;
		lot = new String[w * h];
		Arrays.setAll(lot, (index) -> "free");
	}
	
	public void setLot(String[] newLot) {
		lot = newLot;
		System.out.println("Parking lot has been updated!");
	}
	
	public void setSpot(int w, int h, String spot) {
		lot[h*width + w] = spot;
	}
	
	public String[] getLot() {
		return lot;
	}
	
	public String getSpot(int w, int h) {
		return lot[h*width + w];
	}
	
	public int getWidth() {
		return width;
	}

	public int getHeight() {
		return height;
	}
	
	public String getJson() {
		String lotJson = gson.toJson(lot);
		return "{\"width\": "+width+", \"height\": "+height+", \"parkingLot\": "+lotJson+"}";
	}
}

File: server/src/main/webapp/src/components/ParkingLot.js
import ParkingSpot from "./ParkingSpot";

export default function ParkingLot(props) {

    // Get from server
    let length = props.api.getHeight()
    let width = props.api.getWidth()
    let lot = props.api.getParkingLot()

    let parkingLot = []
    let parkingRow = []

    for (let i = 0; i < length; i++) {
        parkingRow = []
        for (let j = 0; j < width; j++) {
            parkingRow.push(lot[width*i + j])
        }
        parkingLot.push(parkingRow)
    }

    let lotComponent = 
        <div className="parking-lot">
            {parkingLot.map(row => 
                <div className="parking-row">
                    {row.map(spot => 
                        <ParkingSpot spot={spot} />
                    )}
                </div>
            )}
        </div>

    return lotComponent
}